### DISCUSSION TITLE
Race condition allows 2 customer accounts with the same email to created.

### DISCUSSION THREAD

**Discussion by jramoyo-serenade** [swellstores/community#566] [Help]
We create Customer Accounts via events.

When 2 events with the same email are processed at the same time, 2 Customer Accounts get created with the same email instead of 1 request failing due to the unique email constraint. 

This happens intermittently, and I've been able to reproduced it a couple of times.

![Screenshot 2024-08-16 at 9 59 34 am](https://github.com/user-attachments/assets/16cd62c2-2c70-4cdc-9d7d-78878602efbf)

**Comment by logeshswell** [comment]
Yes, it is possible that more than one customer account with the same email could be created at the same time, as we don't have unique constraint at the database level.
